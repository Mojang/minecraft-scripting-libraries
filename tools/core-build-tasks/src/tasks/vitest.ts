import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
