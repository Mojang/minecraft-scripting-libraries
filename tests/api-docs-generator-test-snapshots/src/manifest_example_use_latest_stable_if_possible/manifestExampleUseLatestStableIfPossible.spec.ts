// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Manifest Examples', () => {
    it('Properly generates using latest stable if possible', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['msdocs', 'ts'],
        });
    });
});
