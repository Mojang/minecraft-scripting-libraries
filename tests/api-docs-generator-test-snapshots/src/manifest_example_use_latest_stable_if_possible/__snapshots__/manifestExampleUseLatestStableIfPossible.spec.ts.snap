// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Manifest Examples > Properly generates using latest stable if possible > # Of Files Generated 1`] = `9`;

exports[`Manifest Examples > Properly generates using latest stable if possible > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: test-module1
    items:
    - name: test-module1
      href: test-module1/test-module1.md
    - name: test-module1 Changelog
      href: test-module1/changelog.md
  - name: test-module2
    items:
    - name: test-module2
      href: test-module2/test-module2.md
    - name: test-module2 Changelog
      href: test-module2/changelog.md
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > msdocs/scriptapi/test-module1/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module1 Changelog
description: Changelog of the \`test-module1\` module
---
# \`test-module1\` Changelog

## Version Changes
- [1.1.0-beta](#110-beta)
- [1.0.0](#100)

## 1.1.0-beta
## 1.0.0
#### Added \`test-module1\` Module
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > msdocs/scriptapi/test-module1/test-module1.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module1 Module
description: Contents of the test-module1 module
---
# \`test-module1\` Module

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "test-module1",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.18.20 of Minecraft.

## Available Versions
- \`1.1.0-beta\`
- \`1.0.0\`
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > msdocs/scriptapi/test-module2/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module2 Changelog
description: Changelog of the \`test-module2\` module
---
# \`test-module2\` Changelog

## Version Changes
- [1.1.0-beta](#110-beta)
- [1.0.0-beta](#100-beta)

## 1.1.0-beta
## 1.0.0-beta
#### Added \`test-module2\` Module
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > msdocs/scriptapi/test-module2/test-module2.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: test-module2 Module
description: Contents of the test-module2 module
monikerRange: "=minecraft-bedrock-experimental"
---
# \`test-module2\` Module

> [!CAUTION]
> This module is still in pre-release.  It may change or it may be removed in future releases.

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "test-module2",
    "version": "1.1.0-beta"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.18.20 of Minecraft.

## Available Versions
- \`1.1.0-beta\`
- \`1.0.0-beta\`
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > typescript/test-module1@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module1",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > typescript/test-module1@1.1.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module1",
 *   "version": "1.1.0-beta"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > typescript/test-module2@1.0.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module2",
 *   "version": "1.0.0-beta"
 * }
 * \`\`\`
 *
 */
"
`;

exports[`Manifest Examples > Properly generates using latest stable if possible > typescript/test-module2@1.1.0-beta.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @beta
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "test-module2",
 *   "version": "1.1.0-beta"
 * }
 * \`\`\`
 *
 */
"
`;
